rust/node/src/wallets/wallets.rs

```diff
@@ -1575,6 +1602,31 @@ impl WalletsExt for Arc<Wallets> {
         block
     }
 
+    fn receive_action2(
+        &self,
+        wallet_id: &WalletId,
+        send_hash: BlockHash,
+        representative: PublicKey,
+        amount: Amount,
+        account: Account,
+        work: u64,
+        generate_work: bool,
+    ) -> Result<Option<BlockEnum>, WalletsError> {
+        let guard = self.mutex.lock().unwrap();
+        let wallet = Wallets::get_wallet(&guard, wallet_id)?;
+        
+        let tx = self.env.tx_begin_read();
+        if !wallet.store.valid_password(&tx) {
+            return Err(WalletsError::WalletLocked);
+        }
+
+        if wallet.store.find(&tx, &account.into()).is_end() {
+            return Err(WalletsError::AccountNotFound);
+        }
+
+        Ok(self.receive_action(wallet, send_hash, representative, amount, account, work, generate_work))
+    }
+
```

Comment:
This should be implemented the other way around. receive_action should call receive_action2. Otherwise the checks for wallet locked and account not found would be made twice

Reply:
I removed it and also `change_action2` in #136 , since it was merged.